app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+module Admin
+  class UsersController < ApplicationController
+    before_action :login_required
+
```

Comment:
Why not just use require RestrictAccessToAdminMode?  Just because the redirects are different if denied?

Reply:
Totally doesn't matter.  In fact, I think the user index is not paginated(?) and very expensive (and utterly useless!) so I've been meaning to change it to redirect to / anyway!

If this redirect is the only thing standing in the way of significant simplification, then toss it!

As I see it, and maybe I'm missing something still, we could rip out all of the RestrictAccessToAdminMode stuff and put it in admin_controller.rb to be shared automatically with all admin subcontrollers.  One fewer file, and more standardized admin behavior, and less chance of forgetting to include "require RestrictAccessToAdminMode" in any new admin controllers.